src/openjd/model/v2023_09/_model.py

```diff
@@ -52,8 +52,44 @@
 )
 
 
+class ModelParsingContext:
+    """Context required while parsing an OpenJDModel. An instance of this class
+    must be provided when calling model_validate.
+
+        OpenJDModelSubclass.model_validate(data, context=ModelParsingContext())
+
+    Individual validators receive this value as ValidationInfo.context.
+    """
+
+    extensions: set[str]
+    """Initially, is the set of supported extension names. When the 'extensions'
+    field of the template is processed, this becomes the set of extensions that
+    the the template requested."""
```

Comment:
nit
```suggestion
    extensions: set[str]
    """Initially it is the set of supported extension names. When the 'extensions'
    field of the template is processed, this becomes the set of extensions that
    the template requested."""
```
Aside: is it obvious when this context changes... it's context for users of the library? Feels a little smelly to change the meaning of the variable on the instance?

Also a little weird that this class is public but we're describing behaviour that is more about how we're interacting with the object rather than anything about the object itself

Reply:
It's context for the parsing to use. I'll give this comment more of an overall to explain that.